### ISSUE TITLE
Minio High CPU usage with more than 1000 bucket/user/policy/event

### ISSUE THREAD

**Issue by fdefilippo** [minio/minio#8982]
Using latest Minio: RELEASE.2020-02-07T23-28-16Z and creating more than 1000 bucket/user/canned policy as follow:

```
#!/bin/bash

s3instance="localminio"
s3webhook="arn:minio:sqs::1:webhook"

policy='{"Version": "2012-10-17","Statement":[{"Effect": "Allow","Principal":{"AWS":["*"]},"Action":["s3:GetBucketLocation","s3:ListBucket","s3:ListBucketMultipartUploads"],"Resource":["arn:aws:s3:::BUCKETNAME","arn:aws:s3:::BUCKETNAME/*"]},{"Effect": "Allow","Principal":{"AWS":["*"]},"Action":["s3:AbortMultipartUpload","s3:DeleteObject","s3:GetObject","s3:ListMultipartUploadParts","s3:PutObject"],"Resource":["arn:aws:s3:::BUCKETNAME/in/*","arn:aws:s3:::BUCKETNAME/in/out/*"]}]}'

creates3user(){
    
    mc mb "$s3instance/${accesskey}/in/" 
    mc mb "$s3instance/${accesskey}/out/" 
    
    mc event add "$s3instance/${accesskey}" --event put,delete,get
    
    tmpfile=$(mktemp /var/tmp/XXXXXXXXXXXXXXXXXXX)
    echo ${policy//BUCKETNAME/$accesskey} > ${tmpfile}.json
    mc admin policy add $s3instance policy-rw-${accesskey} ${tmpfile}.json 
    mc admin user add $s3instance ${accesskey} ${secretkey}
    mc admin policy set $s3instance policy-rw-${accesskey} user=$accesskey 
    rm -f ${tmpfile}.json
}

for u in {1..2000}
do
  accesskey="user$u"
  secretkey=$(head -c 10 /dev/urandom | xxd -c 10 -ps)
  creates3user
done 

```
The minio process every 2 minutes takes up 2 cpu for 2 minutes and so on.

**Comment by fdefilippo** [comment]
I may have been unclear, but the use of the cpu is high once the script has finished, minio uses a lot of cpu without anyone using it.

**Comment by kannappanr** [comment]
Closed by #8984
